flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptivebatch/util/AllToAllVertexInputInfoComputerTest.java

```diff
@@ -0,0 +1,313 @@
+                        Map.of(new IndexRange(0, 9), new IndexRange(1, 1)),
+                        Map.of(new IndexRange(0, 9), new IndexRange(2, 2)));
+        checkJobVertexInputInfo(
+                7, leftInputInfos, leftTargetConsumedSubpartitionGroups, vertexInputs);
+        List<Map<IndexRange, IndexRange>> rightTargetConsumedSubpartitionGroups =
+                List.of(
+                        Map.of(new IndexRange(0, 9), new IndexRange(0, 0)),
+                        Map.of(new IndexRange(0, 9), new IndexRange(0, 0)),
+                        Map.of(new IndexRange(0, 9), new IndexRange(0, 0)),
+                        Map.of(new IndexRange(0, 9), new IndexRange(0, 0)),
+                        Map.of(new IndexRange(0, 9), new IndexRange(0, 0)),
+                        Map.of(new IndexRange(0, 9), new IndexRange(1, 1)),
+                        Map.of(new IndexRange(0, 9), new IndexRange(2, 2)));
+        checkJobVertexInputInfo(
+                7, rightInputInfos, rightTargetConsumedSubpartitionGroups, vertexInputs);
+    }
+
+    @Test
+    void testComputeAllSkewedInputNotExistIntraInputKeyCorrelation() {
+        testComputeAllSkewedInputNotExistIntraInputKeyCorrelation(1);
+        testComputeAllSkewedInputNotExistIntraInputKeyCorrelation(10);
+    }
+
+    void testComputeAllSkewedInputNotExistIntraInputKeyCorrelation(int numInputInfos) {
+        AllToAllVertexInputInfoComputer computer = createAllToAllVertexInputInfoComputer();
+        List<BlockingInputInfo> inputInfos = new ArrayList<>();
+        List<BlockingInputInfo> leftInputInfos =
+                createBlockingInputInfos(1, numInputInfos, 2, false, true, List.of(1));
+        List<BlockingInputInfo> rightInputInfos =
+                createBlockingInputInfos(2, numInputInfos, 2, false, true, List.of(1));
+        inputInfos.addAll(leftInputInfos);
+        inputInfos.addAll(rightInputInfos);
+        Map<IntermediateDataSetID, JobVertexInputInfo> vertexInputs =
+                computer.compute(new JobVertexID(), inputInfos, 2, 1, 2);
+
+        List<Map<IndexRange, IndexRange>> leftTargetConsumedSubpartitionGroups =
+                List.of(
+                        Map.of(
+                                new IndexRange(0, 0),
+                                new IndexRange(1, 1),
+                                new IndexRange(0, 1),
+                                new IndexRange(0, 0)),
+                        Map.of(
+                                new IndexRange(1, 1),
+                                new IndexRange(1, 1),
+                                new IndexRange(0, 1),
+                                new IndexRange(2, 2)));
+
+        List<Map<IndexRange, IndexRange>> rightTargetConsumedSubpartitionGroups =
+                List.of(
+                        Map.of(new IndexRange(0, 1), new IndexRange(0, 1)),
+                        Map.of(new IndexRange(0, 1), new IndexRange(1, 2)));
```

Comment:
How about to use some expected results which can better reflects the result? e.g. the expected data size that each will consume from each input